server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/ClusterScheduler.java

```diff
@@ -109,11 +109,11 @@ public void start() {
         return;
       }
     } catch (InterruptedException | ExecutionException e) {
-      // If the dispatch failed, we make the QueryState as failed, and return.
+      // If the dispatch request cannot be sent or TException is caught, we will retry this query.
       if (e instanceof InterruptedException) {
         Thread.currentThread().interrupt();
       }
-      stateMachine.transitionToFailed(e);
+      stateMachine.transitionToRetrying(e);
```

Comment:
In most cases, we won't reach there, if the node is killed, we will get a TException in `dispatchRemote` and then throw a FragmentInstanceDispatchException. Finally,  in `dispatchRead` we will catch that Exception and just return a FragInstanceDispatchResult whose successful is false and failureStatus is the root cause.